config/config_test.go

```diff
@@ -261,3 +262,61 @@ func TestEnabledServers(t *testing.T) {
 		assert.False(t, IsServerEnabled(RegistryType))
 	})
 }
+
+func TestDiscoverFederation(t *testing.T) {
+	viper.Reset()
+	// Server to be a "mock" federation
+	server := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// make our response:
+		response := FederationDiscovery{
+			DirectorEndpoint:              "director",
+			NamespaceRegistrationEndpoint: "registry",
+			JwksUri:                       "jwks",
+			BrokerEndpoint:                "broker",
+		}
+
+		responseJSON, err := json.Marshal(response)
+		if err != nil {
+			http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+			return
+		}
+
+		w.WriteHeader(http.StatusOK)
+		_, err = w.Write(responseJSON)
+		assert.NoError(t, err)
+	}))
+	defer server.Close()
+	t.Run("testInvalidDiscoveryUrlWithPath", func(t *testing.T) {
+		viper.Set("tlsskipverify", true)
+		viper.Set("Federation.DiscoveryUrl", server.URL+"/this/is/some/path")
+		err := DiscoverFederation()
+		assert.Error(t, err)
```

Comment:
It's nice to have our tests not only assert that there _is_ an error, but that the correct error is returned. It's also nice to look at these types of tests as an index of "if I do XXX I get an error that says YYY". Can modify this to use something like `assert.ErrorIs(t TestingT, err, target error, msgAndArgs ...interface{})`?